source/Mlos.Python/mlos/Optimizers/RegressionModels/RegressionEnhancedRandomForestModel.py

```diff
@@ -493,91 +473,81 @@ def transform_x(self, x, what_to_return='x_star'):
           3. Once Lasso/Ridge regression identifies feature dimensions of little predictive value, those dimensions
              are "filtered" from the expanded feature space.
         """
-        x.fillna(value=0, inplace=True)
-        fit_x = self._explode_x(x)
+        # confirm feature_values_pandas_frame contains all expected columns
+        #  if any are missing, impute NaN values
+        missing_column_names = set.difference(set(self.input_dimension_names), set(x.columns.values))
+        for missing_column_name in missing_column_names:
+            x[missing_column_name] = np.NaN
+
+        x_df = self.one_hot_encoder_adapter.project_dataframe(df=x, in_place=False)

```

Comment:
If the this was true in the constructor:
```
self.polynomial_features_adapter = ContinuousToPolynomialBasisHypergridAdapter(adaptee=self.one_hot_encoder_adapter.target,...)
```
Then you'd only need to call:
`x_df = self.polynomial_features_adapter.project_dataframe(df=x_df, in_place=False)`
and the self.one_hot_encoder_adapter would have been called automatically.

Reply:
```suggestion
```